### PR TITLE
iterutil: More ergonomic interface

### PR DESCRIPTION
**Pull request by andrewbaptist**
Replace the Done call in iterutil with Map. This removes a lot of
duplicate code and simplifies the way it is used. There are no
functional changes as a result of this.

Release note: None

### CODE REVIEW

File: pkg/util/iterutil/iterutil.go [hunk 1]
```diff
@@ -28,15 +28,18 @@ var errStopIteration = errors.New("stop iteration")
...
 // 		}
 // 		// continue when nil error
 // 	}
 //
 func StopIteration() error { return errStopIteration }
 
-// Done tells if the error is ErrStopIteration, i.e., should the iteration stop.
-func Done(err error) bool { return errors.Is(err, errStopIteration) }
+// Map the nil if it is StopIteration, or keep the error otherwise
+func Map(err error) error {
```
Comment (postamar) [inline pkg/util/iterutil/iterutil.go:39]: If we want iterutil to be more monadic shouldn't this be a `Filter` instead of a `Map`?

**Review by postamar — APPROVED** [review]
LGTM. Much nicer! Thanks for this.

File: pkg/util/iterutil/iterutil.go (hunk 1 above)
Comment (andrewbaptist) [reply pkg/util/iterutil/iterutil.go:39]: I spent a little time thinking about how to do this, and couldn't come up with a great solution. If anything the original `Done` was much closer to `Filter`. The problem with it in this case, is trying to both exit a loop early as well as return either success or fail.

Let me know if you think I should look at a different alternative or just merge this one in.

**Comment by andrewbaptist** [comment]
bors r=postamar

File: pkg/util/iterutil/iterutil.go (hunk 1 above)
Comment (postamar) [reply pkg/util/iterutil/iterutil.go:39]: Your change is a strict improvement over what there was so I feel very good about it 👍

**Pull request merged**